### DISCUSSION TITLE
16.18.1-rc1

### DISCUSSION THREAD

**Discussion by RoboMWM** [GriefPrevention/GriefPrevention#2014] [General]
This is a pre-release, make some comments in the thread or something if there's an issue or if there are things that were missed that should be included with the official release!

## What's Changed
* Treat arrow-based Lightable ignition as a block change by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1828
* Candles require full trust, remove duplicate candle cake handling by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1827
* Improve claim creation over existing builds by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1839
* Fix blank message being sent to player on death when death messages are disabled by @Jaren8r in https://github.com/TechFortress/GriefPrevention/pull/1880
* Automatically update maxDepth by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1871
* Fix potential NPE visualizing conflict zones by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1854
* Stop people creating nether portals in claimed land by @Elioby in https://github.com/TechFortress/GriefPrevention/pull/1842
* Remove TheUnderground from loadbefore by @Owen1212055 in https://github.com/TechFortress/GriefPrevention/pull/1912
* Prevent theft from water/lava cauldrons. by @RezzedUp in https://github.com/TechFortress/GriefPrevention/pull/1908
* Fix visualization O(n2) performance by @TauCubed in https://github.com/TechFortress/GriefPrevention/pull/2008
* Don't re-render duplicate boundary visualizations by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1987
* Remove `TheUnderground` from softdepend by @bigpresh in https://github.com/TechFortress/GriefPrevention/pull/2010
* Add permission allowing pre-movement chat by @Bobcat00 in https://github.com/TechFortress/GriefPrevention/pull/1996
* Allow Datastore.getClaim(long id) to return subclaims by @akdukaan in https://github.com/TechFortress/GriefPrevention/pull/2011

## New Contributors
* https://github.com/TechFortress/GriefPrevention/pull/1880 by @Jaren8r
* https://github.com/TechFortress/GriefPrevention/pull/1842 by @Elioby
* https://github.com/TechFortress/GriefPrevention/pull/1912 by @Owen1212055
* https://github.com/TechFortress/GriefPrevention/pull/1908 by @RezzedUp
* https://github.com/TechFortress/GriefPrevention/pull/2008 by @TauCubed

**Full Changelog**: https://github.com/TechFortress/GriefPrevention/compare/16.18...16.18.1-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/TechFortress/GriefPrevention/releases/tag/16.18.1-rc1'>16.18.1-rc1</a>.</em>

**Comment by caleopemeraki** [comment]
Hey Robo! Thank you for this awesome plugin! I just wanted to inquire if this most recent post means that a version supporting 1.19 is in the works? I really love this plugin and would love to have an updated version for my server! Thanks for all your hard work! 

Best,
Cal :))

**Reply by RoboMWM** [reply]
Plugin already works in 1.19, I'm just now trying to cut a release to address some minor fixes and a warning from paper.

**Reply by caleopemeraki** [reply]
Fantastic! Thank you so much for your reply and for this plugin! :))

**Comment by Jikoo** [comment]
Hoping we can get #1266, #1931, and #2015 into the next full release.
Would also be good to make a decision on #1991 before it becomes a permanent part of the wall of PRs that might not make it. In the light of other optimizations, is it necessary? Personally I like it as an option for advanced users just in case.

Much as I'd love for #1921 to make it since I'm reasonably happy with it now, I think it's too large a PR to review in any semblance of hurry.

**Reply by TauCu** [reply]
Honestly I think the best way is to just add a timer to PlayerData that limits how fast someone can visualize anything.
Another change would be to sort the claims to visualize by distance squared to the player, and skip them after they reach a limit, say, 100. This way the visualizations will be effectively limited to 100. You'd just want to make sure you always visualize the parent claim.

An issue I see with #1991 is that managers can't see subclaims by default. That would cause issues.

**Reply by Jikoo** [reply]
I do think a cooldown system would be a good thing for GP in general, and definitely the proper full fix. I hadn't considered the fact that it makes it a lot harder for permissiontrusted people to know where they're trusting users.

**Reply by snurre0** [reply]
I like the addition of #1991, it's at least a nice option to have even if the issue is being fixed in another way. That makes it harder for a troll to lag the server with close to zero effort.

**Edit**:
I just re-wrote this whole reply, as I realized I hadn't tested neither the latest commit in #1987 nor #2008 (been a bit busy irl).

I just now tested them separately, and then together, and I must say that together they seems like the perfect solution to #1986 other than a cooldown, which at this point I'm not sure is even necessary.
With #1987 making it impossible to hold down right click to re-render boundaries over and over, and #2008 making the rendering itself insanely more smooth, I think #1986 can be closed at this point.
Thank you all for your effort on this :)

**Reply by Bobcat00** [reply]
> With https://github.com/TechFortress/GriefPrevention/pull/1987 making it impossible to hold down right click to re-render boundaries over and over, and https://github.com/TechFortress/GriefPrevention/pull/2008 making the rendering itself insanely more smooth, I think https://github.com/TechFortress/GriefPrevention/issues/1986 can be closed at this point.

So is the consensus that #1991 is not needed? If that's the case, I'll close it.

**Reply by snurre0** [reply]
For me personally, I don't really need it.
It's always nice to have more options in general, but it sounds like extra features with extra config options was not really wanted at this point, if I read Robo's message correctly.

**Reply by TauCu** [reply]
As I said before, I think there are better alternatives.
As well as some aforementioned issues.

**Comment by Bobcat00** [comment]
#1991 does what it says on the tin, but I hadn't considered that a malicious person could be the one creating all the subclaims. Perhaps @snurre0 could weigh in on how the other visualization change has helped and if further changes are needed.

**Reply by snurre0** [reply]
I just now saw this, after replying to the other thread:
https://github.com/TechFortress/GriefPrevention/discussions/2014#discussioncomment-5256322

**Reply by RoboMWM** [reply]
Pulled in a PR that optimizes these visualization calls, does this mean this is still an issue? I'd like to avoid more config options, especially with regards to features I want to delegate to addons in the future.

**Reply by snurre0** [reply]
This issue was brought up in two different threads in this post, I replied in the other one I linked above.
TLDR: I think the two PR's currently included should be fine for resolving this issue.

**Reply by RoboMWM** [reply]
ya, read through the comments there too lol, I see it now.

**Comment by RoboMWM** [comment]
Anyone have thoughts on my own fix for a minor issue? https://github.com/TechFortress/GriefPrevention/pull/1977

**Reply by Jikoo** [reply]
Someone sitting dogs all over my property is annoying and I should be able to deal with it. A change to allow builders to remove undesirable dogs is good. Part of me says "sitting pets of non-builders can be unsat/led/etc. by builders" would be valid, but that's a very difficult thing to convey to users.
I think it may be the best midground for GP to take, then addons can handle further details.

**Comment by RoboMWM** [comment]
since I don't see anyone complaining about bugs, and hopefully these PRs I pulled don't have any, imma yolo it and cut a release.

Been so long lol let's see if I don't screw up maven
